javascript/src/index.ts

```diff
@@ -590,18 +590,34 @@ export interface WebhookUnbrandedRequiredHeaders {
   "webhook-signature": string;
 }
 
+const SECRET_FORMAT_RAW = "raw";
+
+export interface WebhookOptions {
+  format?: string;
+}
+
 export class Webhook {
   private static prefix = "whsec_";
   private readonly key: Uint8Array;
 
-  constructor(secret: string) {
+  constructor(secret: string | Uint8Array, options?: WebhookOptions) {
     if (!secret) {
       throw new Error("Secret can't be empty.");
     }
-    if (secret.startsWith(Webhook.prefix)) {
+    if (
```

Comment:
These are a lot of unnecessary ifs that check the same thing, it should be:

```typescript
if (options.format === "raw") {
   // .. all of the raw handling
} else {
   // What's currently there
}
```

Reply:
What, something like this?

```javascript
    if (options?.format !== "raw" && secret instanceof String) {
      if (secret.startsWith(Webhook.prefix)) {
        secret = secret.substring(Webhook.prefix.length);
      }
      this.key = base64.decode(secret as string);
    } else {
      if (options?.format === "raw" && secret instanceof Uint8Array) {
        this.key = secret;
      } else if (options?.format === "raw" && secret instanceof String) {
        this.key = Uint8Array.from(secret, (c) => c.charCodeAt(0));
      }
    }
```

We also need the compiler to know that we are always setting this.key to _something_ or it complains.